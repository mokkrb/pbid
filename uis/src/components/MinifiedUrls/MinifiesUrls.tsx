import React from 'react';
import UrlItem from '../UrlItem';
import { Url } from '../../types';

interface MinifiedUrlsProps {
  urls: Url[];
}

const MinifiedUrls = ({ urls }: MinifiedUrlsProps): JSX.Element => {
  return (
    <div data-testid="minifiedUrls">
      <ul>
        {urls.map((url) => (
          <li key={url.originalUrl}>
            <UrlItem url={url} />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default MinifiedUrls;
