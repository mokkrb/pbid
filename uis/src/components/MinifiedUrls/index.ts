import MinifiedUrls from './MinifiesUrls';

export default MinifiedUrls;
