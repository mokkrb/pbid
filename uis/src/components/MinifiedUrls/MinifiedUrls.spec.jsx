import React from 'react';
import { render, screen } from '@testing-library/react';
import MinifiedUrls from './';

const minifiedUrls = [
  {
    id: '123',
    minifiedUrl: 'https://pbid.io/12345678',
    originalUrl: 'https://www.foo.bar',
  },
  {
    id: '321',
    minifiedUrl: 'https://pbid.io/87654321',
    originalUrl: 'https://www.bar.baz',
  },
];

describe('MinifiedUrls', () => {
  it('should render', () => {
    render(<MinifiedUrls urls={minifiedUrls} />);
    expect(screen.getByTestId('minifiedUrls')).toMatchSnapshot();
  });

  it('should handle an empty list', () => {
    render(<MinifiedUrls urls={[]} />);
    expect(screen.queryAllByRole('listitem')).toHaveLength(0);
  });

  it('should display a list of one url', () => {
    render(<MinifiedUrls urls={[minifiedUrls[0]]} />);
    expect(screen.queryAllByRole('listitem')).toHaveLength(1);
  });

  it('should display a list of multiple urls', () => {
    render(<MinifiedUrls urls={minifiedUrls} />);
    expect(screen.queryAllByRole('listitem')).toHaveLength(2);
  });
});
