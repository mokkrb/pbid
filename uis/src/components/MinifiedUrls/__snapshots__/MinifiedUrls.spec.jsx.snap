// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinifiedUrls should render 1`] = `
<div
  data-testid="minifiedUrls"
>
  <ul>
    <li>
      <div
        class="urlItem"
      >
        <span
          class="originalUrl"
        >
          https://www.foo.bar
          :
        </span>
         
        <a
          class="minifiedUrl"
          href="https://pbid.io/12345678"
          rel="noreferrer"
          target="_blank"
        >
          https://pbid.io/12345678
        </a>
      </div>
    </li>
    <li>
      <div
        class="urlItem"
      >
        <span
          class="originalUrl"
        >
          https://www.bar.baz
          :
        </span>
         
        <a
          class="minifiedUrl"
          href="https://pbid.io/87654321"
          rel="noreferrer"
          target="_blank"
        >
          https://pbid.io/87654321
        </a>
      </div>
    </li>
  </ul>
</div>
`;
